docs/source/models/field_ionization_charge_state_prediction.py

```diff
@@ -0,0 +1,260 @@
+import matplotlib as mpl
+from matplotlib import pyplot as plt
+import numpy as np
+import scipy.constants as sc
+
+
+params = {
+    'font.size': 20,
+    'lines.linewidth': 3,
+    'legend.fontsize': 10,
+    'legend.frameon': True,
+    'xtick.labelsize': 20,
+    'ytick.labelsize': 20,
+    # RTD default textwidth: 800px
+    'figure.figsize': [10, 16]
+}
+mpl.rcParams.update(params)
+
+# dictionary of atomic units (AU) - values in SI units
+atomic_unit = {
+    'electric field' : sc.m_e**2 * sc.e**5 / (sc.hbar**4 * (4*sc.pi*sc.epsilon_0)**3),
+    'intensity' : sc.m_e**4 / (8 * sc.pi * sc.alpha * sc.hbar**9) * sc.e**12 / (4 * sc.pi * sc.epsilon_0)**6,
+    'energy' : sc.m_e*sc.e**4/(sc.hbar**2 * (4 * sc.pi * sc.epsilon_0)**2),
+    'time' : sc.hbar**3 * (4 * sc.pi * sc.epsilon_0)**2 / sc.m_e / sc.e**4
+}
+
+# (critical) barrier suppression field strength
+def F_crit(Z,E_Ip):
+    """
+    Classical barrier suppression field strength
+
+
+    :param Z: charge state of the resulting ion
+    :param E_Ip: ionization potential [unit: AU]
+
+    :returns: critical field strength [unit: AU]
+    """
+    return E_Ip**2. / (4. * Z)
+
+def n_eff(Z,E_Ip):
+    """
+    Effective principal quantum number
+
+    :param Z: charge state of the resulting ion
+    :param E_Ip: ionization potential [unit: AU]
+
+    :returns: effective principal quantum number
+    """
+    return Z / np.sqrt(2. * E_Ip)
+
+def ADKRate(Z,E_Ip,F):
+    """
+    Ammosov-Delone-Krainov ionization rate simplified by Stirling's approximation and m=0
+    like in publication [DeloneKrainov1998].
+
+    :param Z: charge state of the resulting ion
+    :param E_Ip: ionization potential [unit: AU]
+    :param F: field strength [unit: AU]
+    """
+    nEff = np.float64(n_eff(Z,E_Ip))
+    D = ((4. * Z**3.) / (F * nEff**4.))**nEff
+
+    rate = (np.sqrt((3. * nEff**3. * F) / (np.pi * Z**3.))
+          * (F * D**2.) / (8. * np.pi * Z)
+          * np.exp(-(2. * Z**3.)/(3. * nEff**3. * F)))
+
+    # set nan values due to near-zero field strengths to zero
+    rate = np.nan_to_num(rate)
+
+    return rate
+
+def convert_a0_to_Intensity(E_in_a0,lambda_laser=800.e-9):
+    """
+    :param E_in_a0: electric field [unit: a0]
+    :param lambda_laser: laser wavelength [unit: m]
+
+    :returns: intensity [unit: W/m²]
+    """
+
+    return (E_in_a0 / (0.85 * lambda_laser * 1e6))**2 * 1e22
```

Comment:
Could you explain the magic numbers?

- `0.85`
- `1e6`
-  `1e22`

or give them named variable names.